From f90aa1b061b9d788833e343563fbac3e7f79ad12 Mon Sep 17 00:00:00 2001
From: Skylar Chang <chiaweic@codeaurora.org>
Date: Tue, 2 Aug 2016 18:16:59 -0700
Subject: msm: gsi: fix memory corruption from debugfs

Fix to GSI reset stats logic to clear the correct stats per channel

Change-Id: I94ae8c769ec2868a291e75a2f47c85bd8a449937
CRs-Fixed: 1046484, 2059810
Acked-by: Ady Abraham <adya@qti.qualcomm.com>
Signed-off-by: Skylar Chang <chiaweic@codeaurora.org>
---
 drivers/platform/msm/gsi/gsi_dbg.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/platform/msm/gsi/gsi_dbg.c b/drivers/platform/msm/gsi/gsi_dbg.c
index bd8cdf3..2ab8b79 100644
--- a/drivers/platform/msm/gsi/gsi_dbg.c
+++ b/drivers/platform/msm/gsi/gsi_dbg.c
@@ -653,11 +653,11 @@ static ssize_t gsi_rst_stats(struct file *file,
 	} else if (ch_id < 0 || ch_id >= GSI_MAX_CHAN ||
 		   !gsi_ctx->chan[ch_id].allocated) {
 		goto error;
+	} else {
+		min = ch_id;
+		max = ch_id + 1;
 	}
 
-	min = ch_id;
-	max = ch_id + 1;
-
 	for (ch_id = min; ch_id < max; ch_id++)
 		memset(&gsi_ctx->chan[ch_id].stats, 0,
 			sizeof(gsi_ctx->chan[ch_id].stats));
-- 
cgit v1.1

